atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -269,27 +344,62 @@ private static SerializableTransactionManager create(
+                    + " live-reloadable atlas configs. Default to using configs specified on the 'atlas' block");
+            return;
+        }
+
+        if (runtimeConfig.enableSweep() == null) {
+            log.warn("Use the 'enableSweep' config on the atlas-runtime block to make it live-reloadable");
+        }
+        if (runtimeConfig.getSweepPauseMillis() == null) {
+            log.warn("Use the 'getSweepPauseMillis' config on the atlas-runtime block to make it live-reloadable");
+        }
+        if (runtimeConfig.getSweepReadLimit() == null) {
+            log.warn("Use the 'getSweepReadLimit' config on the atlas-runtime block to make it live-reloadable");
+        }
+        if (runtimeConfig.getSweepPauseMillis() == null) {
+            log.warn("Use the 'getSweepPauseMillis' config on the atlas-runtime block to make it live-reloadable");
+        }
+        if (runtimeConfig.getSweepDeleteBatchHint() == null) {
+            log.warn("Use the 'getSweepDeleteBatchHint' config on the atlas-runtime block to make it live-reloadable");
+        }
+    }
+
+    private static SweepBatchConfig getSweepBatchConfig(AtlasDbRuntimeConfig runtimeConfig, AtlasDbConfig config) {
         if (config.getSweepBatchSize() != null || config.getSweepCellBatchSize() != null) {
             log.warn("Configuration parameters 'sweepBatchSize' and 'sweepCellBatchSize' have been deprecated"
                     + " in favor of 'sweepMaxCellTsPairsToExamine', 'sweepCandidateBatchSize'"
                     + " and 'sweepDeleteBatchSize'. Please update your configuration files.");
         }
         return ImmutableSweepBatchConfig.builder()
                 .maxCellTsPairsToExamine(chooseBestValue(
+                        runtimeConfig.getSweepReadLimit(),
                         config.getSweepReadLimit(),
                         config.getSweepCellBatchSize(),
                         AtlasDbConstants.DEFAULT_SWEEP_READ_LIMIT))
                 .candidateBatchSize(chooseBestValue(
+                        runtimeConfig.getSweepCandidateBatchHint(),
                         config.getSweepCandidateBatchHint(),
                         config.getSweepBatchSize(),
                         AtlasDbConstants.DEFAULT_SWEEP_CANDIDATE_BATCH_HINT))
-                .deleteBatchSize(MoreObjects.firstNonNull(
+                .deleteBatchSize(chooseBestValue(
+                        runtimeConfig.getSweepDeleteBatchHint(),
                         config.getSweepDeleteBatchHint(),
                         AtlasDbConstants.DEFAULT_SWEEP_DELETE_BATCH_HINT))
                 .build();
     }
 
+    private static int chooseBestValue(@Nullable Integer runtimeOption,
+            @Nullable Integer newOption, @Nullable Integer oldOption, int defaultValue) {
+        if (runtimeOption != null) {
+            return runtimeOption;
+        }
+        return chooseBestValue(newOption, oldOption, defaultValue);
+    }
+
```

Comment:
I actually didn't know this method existed. I was looking only on MoreUtils.firstNonNull. Thanks!